metrics/mahalanobis/mahalanobis.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Mahalanobis metric."""
+
+import numpy as np
+
+import datasets
+
+
+_DESCRIPTION = """
+Compute the Mahalanobis Distance
+
+Mahalonobis distance is the distance between a point and a distribution.
+And not between two distinct points. It is effectively a multivariate equivalent of the Euclidean distance.
+It was introduced by Prof. P. C. Mahalanobis in 1936
+and has been used in various statistical applications ever since
+[source: https://www.machinelearningplus.com/statistics/mahalanobis-distance/]
+"""
+
+_CITATION = """\
+@article{de2000mahalanobis,
+  title={The mahalanobis distance},
+  author={De Maesschalck, Roy and Jouan-Rimbaud, Delphine and Massart, D{\'e}sir{\'e} L},
+  journal={Chemometrics and intelligent laboratory systems},
+  volume={50},
+  number={1},
+  pages={1--18},
+  year={2000},
+  publisher={Elsevier}
+}
+"""
+
+_KWARGS_DESCRIPTION = """
+Args:
+    predictions: Predicted labels, as returned by a model.
+    references: Ground truth labels.
+Returns:
+    matthews_correlation: Matthews correlation.
```

Comment:
(bad copy paste, I guess)
```suggestion
    mahalanobis: The Mahalonobis distance.
```